pydicom/dataset.py

```diff
@@ -181,7 +180,9 @@ def add_new(self, element_offset, VR, value):
             The value of the data element. See :meth:`Dataset.add_new()`
             for a description.
         """
-        self.dataset.add_new(self.get_tag(element_offset), VR, value)
+        tag = self.get_tag(element_offset)
+        self.dataset.add_new(tag, VR, value)
+        self.dataset[tag].private_creator = self.private_creator
```

Comment:
What happens if there's a mix of private creator values?

Reply:
Not sure what you mean - a private block has one private creator, and the data element belongs to it. Can you explain what you mean by a mix of private creator values?

Reply:
I mean suppose a dataset was created with private elements by "PC_01", then opened and modified and more private elements added by "PC_02". What would happen if pydicom then read the file?

Reply:
That's not `Dataset.private_creator`, but `PrivateBlock.private_creator`. There is no private creator in `Dataset`, it holds the private blocks in `_private_blocks`.

Reply:
Ah, so it is.